### PR TITLE
Create smithy-protocol-tests for rpcv2Cbor tests

### PR DESCRIPTION
**Pull request by kstich**
#### Background
* What do these changes do? 
Move the RPCv2 CBOR tests to a new package.
* Why are they important?
The protocol is not AWS specific, so these tests belong in their own package for clean dependency separation.

#### Testing
* How did you test these changes?
Build + check jar files.

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: smithy-aws-protocol-tests/model/shared-types.smithy [hunk 1]
```diff
@@ -121,10 +121,6 @@ list LongSet {
     member: Long,
 }
 
-list TimestampList {
```
Comment (kstich) [inline smithy-aws-protocol-tests/model/shared-types.smithy:124]: This was a duplicate definition that was also cleaned up in the `smithy-protocol-tests` copy.

File: smithy-protocol-tests/build.gradle [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+plugins {
+    id "software.amazon.smithy.gradle.smithy-jar" version "$smithyGradleVersion"
+}
+
+description = "Defines protocol tests for Smithy HTTP protocols."
```
Comment (hpmellema) [inline smithy-protocol-tests/build.gradle:10]: RPCv2?

File: smithy-protocol-tests/build.gradle [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+ext {
+    displayName = "Smithy :: Protocol Tests"
```
Comment (hpmellema) [inline smithy-protocol-tests/build.gradle:13]: Should this be updated to be specific to cbor? Same with the modulename.

File: smithy-protocol-tests/build.gradle (hunk 2 above)
Comment (kstich) [reply smithy-protocol-tests/build.gradle:10]: Package is not specific to RPCv2, even if it only contains that for now.

File: smithy-protocol-tests/build.gradle (hunk 3 above)
Comment (kstich) [reply smithy-protocol-tests/build.gradle:13]: These aren't specific to RPCv2 CBOR.

**Review by hpmellema — APPROVED** [review]

**Pull request merged**